proxy/proxy.go

```diff
@@ -31,5 +32,31 @@ func (p *Proxy) WithTransport(transport *http.Transport) *Proxy {
 func (p *Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	log.Println(r.URL)
 	r.Host = p.target.Host
+	r.Header.Set("X-Forwarded-For", r.RemoteAddr) // Note: not very reliable. Maybe we should set and check X-Real-IP. Also, X-Forwarded-For might have already been set.
```

Comment:
Because TLS terminates here and not in i.e. Cloudflare Proxy or an ALB, `RemoteAddr` should work. At least for now that seems good enough. Which hop would add another source IP header?

Reply:
Perhaps if we'd do TCP load balancing with HAProxy it would be a different IP 🤔

Reply:
Yes, the RemoteAddr can change -- there's the X-Real-IP header we could be populating
But I'm not certain it will change because as you say we are terminating TLS here

Reply:
oh right. i guess this one is good to start.

one other to note about reusing `X-Forwarded-For` is that it can be a list of IPs of all the hops (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For) and the first entry would be the original one.

i think we can leave as-is for now and reevaluate later when we need it.

Reply:
Turns out the proxy itself already populates X-Forwarded-For, how convenient is that